src/screen/KeyList/index.tsx

```diff
@@ -1,73 +1,152 @@
-import React, { memo } from 'react'
-import { FiFolder } from 'react-icons/fi'
+import React, {
+  memo,
+  useState,
+  useEffect,
+  useRef,
+  useCallback,
+  ChangeEvent,
+  useMemo
+} from 'react'
+import { useDebounce } from 'react-use'
+import { useVirtual } from 'react-virtual'
 
+import { useRecoilState } from 'recoil'
+
+import {
+  currentConnectionState,
+  currentDatabaseState,
+  currentKeyState
+} from '../../atoms/connections'
+import EmptyContent from '../../components/EmptyContent'
 import { useWindowSize } from '../../hooks/useWindowSize'
+import { loadKeysFromDatabase } from '../../services/database/LoadKeysFromDatabase'
 import SearchInput from './SearchInput'
 import {
   Container,
   Header,
   HeaderTitle,
   HeaderTextContainer,
-  HeaderTotalKeys,
+  HeaderDatabaseDetails,
+  KeyListWrapper,
   KeyListContainer,
   Key,
   KeyTextContainer,
-  KeyTitle,
-  KeyChidrenCount
+  KeyTitle
 } from './styles'
 
 const KeyList: React.FC = () => {
+  const parentRef = useRef(null)
+
   const { width } = useWindowSize({ watch: false })
 
+  const [searchInputValue, setSearchInputValue] = useState('')
+  const [filter, setFilter] = useState('')
+  const [keys, setKeys] = useState<string[]>([])
+
+  const [currentConnection] = useRecoilState(currentConnectionState)
+  const [currentDatabase] = useRecoilState(currentDatabaseState)
+  const [currentKey, setCurrentKey] = useRecoilState(currentKeyState)
+
+  useDebounce(
+    () => {
+      setFilter(searchInputValue)
+    },
+    500,
+    [searchInputValue]
+  )
+
+  const filteredKeys = useMemo(() => {
+    if (!filter) {
+      return keys
+    }
+
+    return keys.filter(key => key.includes(filter))
+  }, [filter, keys])
+
+  const rowVirtualizer = useVirtual({
+    size: filteredKeys.length,
+    parentRef,
+    estimateSize: useCallback(() => 33, [])
+  })
+
+  const handleSearchInputChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setSearchInputValue(e.target.value)
+    },
+    []
+  )
+
+  const handleSelectKey = useCallback((key: string) => {
+    setCurrentKey(key)
+  }, [])
+
+  useEffect(() => {
+    if (currentDatabase) {
```

Comment:
I would like add 
```
setCurrentKey(undefined)
```
above to reset content value, its triggering me, but maybe its a feature

Reply:
Fixed.